### PR TITLE
[MVP] Add player light cycle movement

### LINKED ISSUE #7: Player Light Cycle Movement
Description:
Implement the logic for controlling the movement of the light cycle.

Acceptance Criteria:

    Light cycle can move in four directions: up, down, left, and right.
    Light cycle leaves a trail behind as it moves.

### PR DESCRIPTION
**Pull request by gsavchenko**
resolves #7 

## Overview
This PR adds the ability to control a 'light cycle' using keyboard controls.

## Changes
- Installs @react-three/drei
  - This is helpful library that includes utilities for working with react-three-fiber
-  Adds a "Grid" component as a background
- Removes index from "features"
  - We should be accessing features subfolders instead of exporting them all from route
- Creates "Player" feature that has the controls passed down to the "PlayerMesh"
- Removes a bunch of next specific files and styles

## Demo

https://github.com/gsavchenko/MazeFaze/assets/9406280/2780ebe6-ab7c-4ce5-989d-ec3e1b757282

### CODE REVIEW

File: mf_frontend/src/app/components/three/index.ts
Comment (benxia-z) [inline mf_frontend/src/app/components/three/index.ts:1]: Are you going to use this file?

**Review by benxia-z — APPROVED** [review]
Nice!

File: mf_frontend/src/app/components/three/index.ts
Comment (benxia-z) [reply mf_frontend/src/app/components/three/index.ts:1]: For context, this file is empty

File: mf_frontend/src/app/components/three/grid.component.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { useFrame } from "@react-three/fiber";
+
+export const Grid = () => {
+  return (
+    <mesh position={[0, 0, 0]}>
+      <planeGeometry args={[100, 50, 5, 5]} />
```
Comment (mikalcallahan) [inline mf_frontend/src/app/components/three/grid.component.tsx]: Use an interface for the `args` :)

File: mf_frontend/src/features/player/player.component.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export const Player: React.FC = () => {
+  const controls = useMemo<KeyboardControlsEntry<Controls>[]>(
+    () => [
+      { name: Controls.forward, keys: ["ArrowUp", "KeyW"] },
+      { name: Controls.back, keys: ["ArrowDown", "KeyS"] },
+      { name: Controls.left, keys: ["ArrowLeft", "KeyA"] },
+      { name: Controls.right, keys: ["ArrowRight", "KeyD"] },
+    ],
+    [],
+  );
```
Comment (mikalcallahan) [inline mf_frontend/src/features/player/player.component.tsx]: Reconsider scope

File: mf_frontend/src/features/player/player.component.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    setPlayerDirection(event);
+  }, []);
+
+  return (
+    <KeyboardControls
+      map={controls}
+      onChange={(event) => handleKeyboardChange(event as Controls)}
+    >
+      <PlayerMesh direction={playerDirection} />
+    </KeyboardControls>
```
Comment (mikalcallahan) [inline mf_frontend/src/features/player/player.component.tsx:34]: Reconsider scope

**Pull request merged**